app/src/ui/rebase/warn-force-push-dialog.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react'
+
+import { Repository } from '../../models/repository'
+import { WarnForcePushStep } from '../../models/rebase-flow-step'
+
+import { Button } from '../lib/button'
+import { ButtonGroup } from '../lib/button-group'
+import { Checkbox, CheckboxValue } from '../lib/checkbox'
+
+import { Dispatcher } from '../dispatcher'
+import { DialogFooter, DialogContent, Dialog } from '../dialog'
+
+interface IWarnForcePushProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly step: WarnForcePushStep
+  readonly askForConfirmationOnForcePush: boolean
+  readonly onDismissed: () => void
+}
+
+interface IWarnForcePushState {
+  readonly askForConfirmationOnForcePush: boolean
+}
+
+export class WarnForcePushDialog extends React.Component<
+  IWarnForcePushProps,
+  IWarnForcePushState
+> {
+  public constructor(props: IWarnForcePushProps) {
+    super(props)
+
+    this.state = {
+      askForConfirmationOnForcePush: props.askForConfirmationOnForcePush,
+    }
+  }
+
+  public render() {
+    const { baseBranch, targetBranch } = this.props.step
+
+    return (
+      <Dialog
+        title="This rebase will require a force push"
```

Comment:
Could we keep the title as short as possible and also deal with platform casing? Perhaps something like

```ts
const title = __DARWIN__
  ? 'Rebase Will Require Force Push'
  : 'Rebase will require force push'
```